utilities/contribution_utilities.py

```diff
@@ -0,0 +1,28 @@
+from collections import defaultdict
+from typing import Dict
+
+
+def calculate_leaders(contribution_metadata) -> Dict[str, int]:
+    """
+    Takes three lists (from solidity arrays) and returns a dictionary of leaders.
+    """
+    contributors = contribution_metadata[0]
+    amounts = contribution_metadata[1]
+    combined = contribution_metadata[2]
+    datetime = contribution_metadata[3]
+
+    contributors_by_address = defaultdict(list)
+
+    for counter, contributor in enumerate(contributors):
+        is_combined = combined[counter]
+        amount = amounts[counter]
+
+        if is_combined:
+            if not contributors_by_address[contributor]:
+                # This ought to be an impossible situaiton - how did they dcombine with a bid that didn't exist?
```

Comment:
Right now you can call `contributeAndCombine` function of the contract as your first contribution, without previously calling `contribute`, so it's not impossible?